addons/udes_stock/models/stock_picking.py

```diff
@@ -1159,16 +1159,36 @@ def action_confirm(self):
         self.unlink_empty()
         return res
 
+    def find_empty_pickings_to_unlink(self):
+        """ Finds empty pickings to unlink in self, when it is set, otherwise
+            searches for any empty picking.
+        """
+        if self:
+            records = self.filtered(
+                lambda p:
+                    (len(p.move_lines) == 0
+                    and not p.u_mark
+                    and p.is_locked))
+        else:
+            records = self.search([
+                ('u_mark', '=', False),
```

Comment:
now that we're doing this query, it could be worth adding an index on `u_mark`

Reply:
Haven't checked, but will add it if it is not there, thanks.